### ISSUE TITLE
(Focal) No sync on WiFi, when sync on mobile is not allowed

### ISSUE THREAD

**Issue by luksus42** [belohoub/UBsync#28]
When the service got started, while cellular data was active in the network-indicator, it is not possible to (manually) initiate a sync, even if WiFi got activated meanwhile.

The Owncloudsynd log:
```
[07.04.23 14:02] Network Connection Type: "Unknown"
[07.04.23 14:02] Mobile Data Sync:  false
[07.04.23 14:02] No Sync on Mobile Data - Check User Settings - Unable to Sync```
```

Further testing reveals, that syncing with mobile data does not seem to work at all. Even if is allowed in settings.
In case of active mobile data (network-manager) the Network Connection Type is always "Unknown".
In case of WiFi I already spotted two different types: "Ethernet" and "WLAN".

**Comment by belohoub** [comment]
This is strange. It tends to remember the state when OwncloudSyncd started...

"Unknown" is sometimes reported after switching ON/OFF when  OwncloudSyncd is running.

This might be even bug in QT :-(

Further investigation is needed.

**Comment by belohoub** [comment]
@luksus42 please try the latest [pre-release 0.8.0-RC1](https://github.com/belohoub/UBsync/releases/tag/0.8.0-RC1).

It should work now :-)

**Comment by luksus42** [comment]
Hey, regarding WiFi it works now. :+1: 
As soon as I turn on WiFi, syncing works, regardless of the prior state, when the service was started.

Two findings:
1. Even if the fixed service already worked, I needed to remove the old service file, for the app to recognize the new service version (0.8). So this is not really an issue, but I was not sure if the service file or its content got really replaced.
2. Syncing on mobile-data is still not working at all for me, no matter whether the service got started with only mobile data activated or not. In Settings "mobile-sync" is allowed.
   
   ```
   [12.04.23 10:36] Mobile Data Sync:  false
   [12.04.23 10:36] Network Connection  "lo"  Type:  QNetworkInterface::Loopback
   [12.04.23 10:36] Network Connection  "rmnet_ipa0"  Type:  QNetworkInterface::Unknown
   [12.04.23 10:36] Network Connection  "rmnet_data1"  Type:  QNetworkInterface::Unknown
   [12.04.23 10:36] No Sync on Mobile Data - Check User Settings - Unable to Sync.
   [12.04.23 10:36] OwncloudSyncd::syncTargets() - m_syncing = false
   ```

**Comment by luksus42** [comment]
Regarding Finding 2: "Syncing on mobile data":

I now removed the complete click application folder from "/opt/click..." and also the service file from `.config/systemd/user/` (after stopping and disabling it).
Then I installed the latest click new and now mobile data sync is working!
Even switching the mobile sync setting on and off, is recognized the correct way now by the sync-service.

So maybe there were some incomplete file-updates within the multiple manual click installs over the old click versions.
